renku/core/management/datasets.py

```diff
@@ -623,6 +624,9 @@ def _fetch_lfs_files(self, repo_path, paths):
     def _fetch_files_metadata(client, paths):
         """Return metadata for files from paths."""
         files = {}
+
+        migrate(client)
```

Comment:
Good point! However, we should not block users from adding data from a newer version. I'll add this check and return immediately instead of failing. Preferably, we should print a warning message but that requires lots of changes.